### PR TITLE
chang to fluent Interface

### PR DESCRIPTION
**Pull request by diridari**
changed to AssertJ fluent Interface
: EclipseParserTest,ErlcParserTest,FlexSDKParserTest,FxcopParserTest

### CODE REVIEW

File: src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java [hunk 1]
```diff
@@ -21,14 +26,13 @@
...
      * @see <a href="http://issues.jenkins-ci.org/browse/JENKINS-21377">Issue 21377</a>
      */
     @Test
-    public void issue21377() throws IOException {
+    public void issue21377() {
         Issues warnings = createParser().parse(openFile("issue21377.txt"));
-
-        assertEquals(1, warnings.size());
+        assertThat(warnings.size()).isEqualTo(1);
+        //assertEquals(1, warnings.size());
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java]: nix aaskommentiertes Comitten

File: src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java [hunk 2]
```diff
@@ -21,14 +26,13 @@
...
-     * @throws IOException if the file could not be read
      * @see <a href="http://issues.jenkins-ci.org/browse/JENKINS-21377">Issue 21377</a>
      */
     @Test
-    public void issue21377() throws IOException {
+    public void issue21377() {
         Issues warnings = createParser().parse(openFile("issue21377.txt"));
-
-        assertEquals(1, warnings.size());
+        assertThat(warnings.size()).isEqualTo(1);
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java]: assertThat(warnings).hasSize(1)

File: src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java [hunk 3]
```diff
@@ -41,130 +45,143 @@ public void issue21377() throws IOException {
...
-                391,
-                "The method getTableValues(PropertyRestrictionType) from the type PmModelImporter is never used locally",
-                "/media/ssd/multi-x-processor/workspace/msgPM_Access/com.faktorzehn.pa2msgpm.core/src/com/faktorzehn/pa2msgpm/core/loader/PmModelImporter.java",
-                TYPE, CATEGORY, Priority.NORMAL);
-        checkWarning(iterator.next(),
-                56,
-                "The value of the field PropertyImporterTest.ERROR_RESPONSE is not used",
-                "/media/ssd/multi-x-processor/workspace/msgPM_Access/com.faktorzehn.pa2msgpm.core.test/src/com/faktorzehn/pa2msgpm/core/importer/PropertyImporterTest.java",
-                TYPE, CATEGORY, Priority.NORMAL);
+        IssueAssert.assertThat(iterator.next()).hasPriority(Priority.NORMAL)
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java]: Soft Assertions nutzen

File: src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java [hunk 4]
```diff
@@ -41,130 +45,143 @@ public void issue21377() throws IOException {
...
-     * @throws IOException if the file could not be read
      * @see <a href="http://issues.jenkins-ci.org/browse/JENKINS-12822">Issue 12822</a>
      */
     @Test
-    public void issue12822() throws IOException {
+    public void issue12822() {
         Issues warnings = createParser().parse(openFile("issue12822.txt"));
 
-        assertEquals(15, warnings.size());
+        assertThat(warnings.size()).isEqualTo(15);
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java]: s.o.

File: src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java [hunk 5]
```diff
@@ -41,130 +45,143 @@ public void issue21377() throws IOException {
...
+        assertThat(warnings.size()).isEqualTo(8);
 
         for (Issue annotation : warnings) {
             checkWithAnnotation(annotation);
         }
     }
 
     private void checkWithAnnotation(final Issue annotation) {
-        assertTrue(annotation.getMessage().matches("[a-zA-Z].*"), "Wrong first character in message");
+        assertThat(annotation.getMessage().matches("[a-zA-Z].*")).isTrue().withFailMessage("Wrong first character in message");
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java]: Custom assertions

File: src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java [hunk 6]
```diff
@@ -41,130 +45,143 @@ public void issue21377() throws IOException {
...
-                TYPE, CATEGORY, Priority.NORMAL);
+        IssueAssert.assertThat(annotation).hasPriority(Priority.NORMAL)
+                .hasCategory(CATEGORY)
+                .hasLineStart(10)
+                .hasLineEnd(10)
+                .hasMessage("The import com.bombardier.oldinfra.export.dataAccess.InfrastructureDiagramAPI is never used")
+                .hasFileName("/srv/hudson/workspace/Ebitool Trunk/build/plugins/com.bombardier.oldInfra.export.jet/jet2java/org/eclipse/jet/compiled/_jet_infraSoe.java")
+                .hasType(TYPE);
+
     }
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java]: s.o.

File: src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java (hunk 5 above)
Comment (diridari) [reply src/test/java/edu/hm/hafner/analysis/parser/EclipseParserTest.java]: I've created a additional Custom Assertion

**Pull request merged**